sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/postgresql.yaml

```diff
@@ -30,7 +30,18 @@ provider:
           # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
           # Note: Anything matching values like "${DB_PASSWORD:-postgres}" will be ignored and processed by another
           # rule.
-          patternNot: "(?i)^(\\$[a-z_]*)?(db|my)?_?pass(word|wd)?|^\\[[\\w\\t \\-]+\\]$|^<[\\w\\t \\-]+>$|^%s$|^\\{+[^}]*\\}+$|^\\$\\(\\w+\\)$|\\$\\{\\w+(:-\\w+)?\\}"
+          # The last line was added to this pattern to sort out any code that loads values form environment variables
+          # e.g. os.getenv('Password'), os.environ["password"], env("Password")
```

Comment:
I suggest removing all these lines to provide examples, as well as the why of patternNot. Such as :

``` yaml
          # Potential FPs here are environment variables, templates and string substitutions, such as
          # - $my_password
          # - [My Password]
          # - <My Password>
          # - %s
          # - {{MyPassword}}
          # - $(mypassword)
          # - ${mypassword}
          # - getenv("PASS")
```

To do so, just use chatgpt with this prompt:

```
generate a string that would satisfy a given regex without telling anything else. say YES if you understand the assignment.
```

and then send the big regex with markdown backticks, it explains everything

Then, ensure that everything is actually right and add them as comments. I think this way it's gonna be very clear and great for maintainability 👍